File: backend/app/schemas/menu.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class AsyncRouteItem(BaseModel):
    """
    动态路由菜单项
    字段与前端 useRouteStore.AsyncRouteItem 保持一致
    """

    model_config = ConfigDict(populate_by_name=True)

    activeMenu: str = ""
    alwaysShow: bool = False
    breadcrumb: bool = True
    children: list["AsyncRouteItem"] = Field(default_factory=list)
    component: str = ""
    hidden: bool = False
    icon: str = ""
    id: str
    keepAlive: bool = False
    parentId: str = ""
    path: str
    permission: str = ""
    redirect: str = ""
    roles: list[str] = Field(default_factory=list)
    showInTabs: bool = True
    sort: int = 0
    status: Literal["0", "1"] = "1"
    title: str
    type: Literal[1, 2, 3] = 2
    affix: bool = False


AsyncRouteItem.model_rebuild()


class MenuRoutesResponse(BaseModel):
    code: int = 200
    message: str = "success"
    data: list[AsyncRouteItem]

File: frontend/web/src/apis/menu.ts
import type { AsyncRouteItem } from '@/core/stores/useRouteStore'
import { request } from './request'

export type MenuType = 1 | 2 | 3
export type StatusValue = '0' | '1'

export interface MenuItem {
  id: string
  parentId: string
  type: MenuType
  title: string
  path?: string
  component?: string
  redirect?: string
  icon?: string
  permission?: string
  sort?: number
  status: StatusValue
  hidden?: boolean
  keepAlive?: boolean
  affix?: boolean
  alwaysShow?: boolean
  breadcrumb?: boolean
  showInTabs?: boolean
  activeMenu?: string
  isSystem?: boolean
  children?: MenuItem[]
}

export interface MenuFormData {
  parentId: string
  type: MenuType
  title: string
  path: string
  component: string
  redirect: string
  icon: string
  permission: string
  sort: number
  status: StatusValue
  hidden: boolean
  keepAlive: boolean
  affix: boolean
  alwaysShow: boolean
  breadcrumb: boolean
  showInTabs: boolean
  activeMenu: string
}

/** 获取动态路由菜单 */
export function getRoutesApi() {
  return request<AsyncRouteItem[]>({ url: '/menu/routes', method: 'get' })
}

/** 管理端菜单树 */
export function getMenuTreeApi() {
  return request<MenuItem[]>({ url: '/menu/tree', method: 'get' })
}

export function createMenuApi(data: Partial<MenuFormData>) {
  return request({ url: '/menu', method: 'post', data })
}

export function updateMenuApi(id: string, data: Partial<MenuFormData>) {
  return request({ url: `/menu/${id}`, method: 'put', data })
}

export function deleteMenuApi(ids: string[]) {
  return request({ url: '/menu/delete', method: 'post', data: { ids } })
}
